### DISCUSSION TITLE
It is possible to make a multiplayer game, simple 2 players

### DISCUSSION THREAD

**Discussion by MiqueiasDevGames** [GodotNuts/GodotFirebase#337] [General]
I'm trying to use:

`.update(data.key, {})`

to send updates and the patch_data_updat signal, but to no avail.

it is very slow, nothing is realtime and I still have a bunch of:
  **request: HTTPRequest is processing a request. Wait for completion or cancel it before attempting a new one.**

Help me there, how to use the realtime database correctly to have the real time and stop giving this error

**Comment by WolfgangSenff** [comment]
You cannot do it - the developers of Firebase (the backend technology, not us!) have said it is not appropriate for realtime gameplay and you will never achieve the type of speeds you need for that. They do not support it, and they definitely do not recommend it.

**Reply by MiqueiasDevGames** [reply]
> @WolfgangSenff
> 
> Like this? so how did he get it? Link: [https://youtu.be/xhURh2RDzzg[https://youtu.be/xhURh2RDzzg](https://youtu.be/xhURh2RDzzg)](https://youtu.be/xhURh2RDzzg%5Bhttps://youtu.be/xhURh2RDzzg%5D(https://youtu.be/xhURh2RDzzg))

**Comment by MiqueiasDevGames** [comment]
@WolfgangSenff 

Like this? so how did he get it?
Link: https://youtu.be/xhURh2RDzzg[https://youtu.be/xhURh2RDzzg](https://youtu.be/xhURh2RDzzg)

**Reply by WolfgangSenff** [reply]
It's not going to be a very good game, his isn't. Notice there are pauses after moving and stuff because it's actually very slow. What I'd suggest, and what I generally do, is not use update, but push: I make action types and push them to the database and everyone gets those and moves whatever avatar represents the player that made the request. You can even make it so the player doesn't directly make the move, but waits for Firebase to return the push notification to them before moving. But in the end, it's going to be *expensive* to make that game. Even very few players can lead to thousands of requests a minute, and by that point, you should just setup your own server and use Godot's built-in multiplayer code. It's very easy to set that up in 4.0.

All that said, what I said was the truth: I've talked to the Firebase program managers and all have said the same: it isn't appropriate for any kind of real-time gameplay. For turn-based, it's fine, even great, but not for real-time.

**Reply by MiqueiasDevGames** [reply]
@WolfgangSenff 
Have you tried to use realtime database as signaling server together with webRTC GDextesion?

**Comment by WolfgangSenff** [comment]
I have not - I do wonder about it, though.

**Reply by MiqueiasDevGames** [reply]
I am going to try.
